File: app/src/main/java/com/asfoundation/wallet/ui/iab/database/AppCoinsOperationDao.java
package com.asfoundation.wallet.ui.iab.database;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import io.reactivex.Flowable;
import java.util.List;

@Dao public interface AppCoinsOperationDao {
  @Query("select * from AppCoinsOperationEntity")
  Flowable<List<AppCoinsOperationEntity>> getAllAsFlowable();

  @Query("select * from AppCoinsOperationEntity") List<AppCoinsOperationEntity> getAll();

  @Query("select * from AppCoinsOperationEntity where `key` like :key limit 1")
  Flowable<AppCoinsOperationEntity> getAsFlowable(String key);

  @Query("select * from AppCoinsOperationEntity where `key` like :key limit 1")
  AppCoinsOperationEntity get(String key);

  @Insert void insert(AppCoinsOperationEntity data);

  @Delete void delete(AppCoinsOperationEntity data);
}

File: app/src/main/java/com/asfoundation/wallet/ui/iab/database/AppCoinsOperationDao.kt
package com.asfoundation.wallet.ui.iab.database

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import io.reactivex.Flowable

@Dao
interface AppCoinsOperationDao {
  @Query("select * from AppCoinsOperationEntity")
  fun getAllAsFlowable(): Flowable<List<AppCoinsOperationEntity>>

  @Query("select * from AppCoinsOperationEntity")
  fun getAll(): List<AppCoinsOperationEntity>

  @Query("select * from AppCoinsOperationEntity where `key` like :key limit 1")
  fun getAsFlowable(key: String): Flowable<AppCoinsOperationEntity>

  @Query("select * from AppCoinsOperationEntity where `key` like :key limit 1")
  fun get(key: String): AppCoinsOperationEntity

  @Insert
  fun insert(data: AppCoinsOperationEntity)

  @Delete
  fun delete(data: AppCoinsOperationEntity)
}
